src/librustc_span/symbol.rs

```diff
@@ -980,6 +980,31 @@ impl fmt::Display for IdentPrinter {
     }
 }
 
+/// An newtype around `Ident` that calls [Ident::modern_and_legacy] on
+/// construction.
+// FIXME(matthewj, petrochenkov) Use this more often, add a similar
+// `ModernIdent` struct and use that as well.
+#[derive(Copy, Clone, Eq, PartialEq, Hash)]
+pub struct LegacyIdent(Ident);
```

Comment:
These (`LegacyIdent`, `modern`, `modern_and_legacy`) are one of the hardest cases of naming that I've encountered in rustc.
The names are unambiguously bad, but I can't come up with anything better now.

I need to think how to rename them in some less confusing and more consistent way.